crates/index-scheduler/src/lib.rs

```diff
@@ -922,6 +927,22 @@ impl IndexScheduler {
     }
 }
 
+fn configure_experimental_features(
+    env: &Env<WithoutTls>,
+    features: &FeatureData,
+    options: &IndexSchedulerOptions,
+) -> Result<()> {
+    let current_features = features.features().runtime_features();
+
+    let new_features = RuntimeTogglableFeatures {
+        chat_completions: options.experimental_chat_completions,
+        ..current_features
+    };
+
+    let wtxn = env.write_txn().map_err(Error::HeedTransaction)?;
+    features.put_runtime_features(wtxn, new_features)
```

Comment:
This method accepts `mut`, not `&mut`, and this this is the reason to create a new `write_txn` instead of using the one at line 287.